### PR TITLE
Fix clang13 build error (facebook#9374)

### LINKED ISSUE #12204: TiKV fail to compile on MacOS
### What version of TiKV are you using?
master

### What operating system and CPU are you using?
MacOS, M1

### Steps to reproduce
make dev

### What did you expect?
Compile success

### What did happened?
```
  /.cargo/git/checkouts/rust-rocksdb-a9a28e74c6ead8ef/651a5c0/librocksdb_sys/libtitan_sys/titan/src/db_impl.cc:100:14: error: variable 'file_size' set but not used [-Werror,-Wunused-but-set-variable]
      uint64_t file_size = 0;
               ^
  1 error generated.
  make[5]: *** [CMakeFiles/titan.dir/src/db_impl.cc.o] Error 1
  make[5]: *** Waiting for unfinished jobs....
  make[4]: *** [CMakeFiles/titan.dir/all] Error 2
  make[3]: *** [CMakeFiles/titan.dir/rule] Error 2
  make[2]: *** [titan] Error 2
  thread 'main' panicked at '
  command did not execute successfully, got: exit status: 2
```

```
% gcc --version
Apple clang version 13.1.6 (clang-1316.0.21.2)
Target: arm64-apple-darwin21.4.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

### PR DESCRIPTION
**Pull request by tabokie**
Signed-off-by: tabokie <xy.tao@outlook.com>

Fix https://github.com/tikv/tikv/issues/12204

### CODE REVIEW

**Comment by BusyJay** [comment]
There is still one error:
```
/workspace/tirocks/tirocks-sys/rocksdb/db/db_impl/db_impl_write.cc:112:12: error: variable 'valid_batches' set but not used [-Werror,-Wunused-but-set-variable]
      size_t valid_batches = 0;
             ^
  1 error generated.
```

File: options/db_options.h [hunk 1]
```diff
@@ -89,6 +89,7 @@ struct MutableDBOptions {
   MutableDBOptions();
   explicit MutableDBOptions(const MutableDBOptions& options) = default;
   explicit MutableDBOptions(const DBOptions& options);
+  MutableDBOptions& operator=(const MutableDBOptions& options) = default;
```
Comment (BusyJay) [inline options/db_options.h]: So why defining a default copy constructor in the first place?

Comment (tabokie) [reply options/db_options.h]: Actually, my fix here is not correct. I should've removed the uses of implicit assignment instead.

The `explicit MutableDBOptions(const MutableDBOptions& options)` is probably here to make creating copy of DB's mutable options less convenient, so we don't end up with lots of options with differing values.

Comment (tabokie) [reply options/db_options.h]: BTW, the `size_t valid_batches = 0;` error is gone in the latest code.

File: options/db_options.h [hunk 2]
```diff
@@ -89,6 +89,7 @@ struct MutableDBOptions {
   MutableDBOptions();
   explicit MutableDBOptions(const MutableDBOptions& options) = default;
   explicit MutableDBOptions(const DBOptions& options);
+  MutableDBOptions& operator=(MutableDBOptions&& options) = default;
```
Comment (BusyJay) [inline options/db_options.h]: I download the latest rocksdb code and can build it without any errors. It seems they delete the explicit copy constructor instead.

Comment (BusyJay) [reply options/db_options.h]: The commit that introduces the change: https://github.com/facebook/rocksdb/commit/9c6fb26033fe83ff8b931b72dc9d1f8496ad5ebd

Comment (tabokie) [reply options/db_options.h]: Ok, I'll cherry-pick it instead.

**Review by Connor1996 — APPROVED** [review]
LGTM

**Review by BusyJay — APPROVED** [review]

**Pull request merged**